src/wallet/rpcnames.cpp

```diff
@@ -676,6 +677,144 @@ name_update (const JSONRPCRequest& request)
 
 /* ************************************************************************** */
 
+UniValue
+queuerawtransaction (const JSONRPCRequest& request)
+{
+  std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest (request);
+  CWallet* const pwallet = wallet.get ();
+
+  if (!EnsureWalletIsAvailable (pwallet, request.fHelp))
+    return NullUniValue;
+
+  if (request.fHelp || request.params.size () < 1 || request.params.size () > 2)
+    throw std::runtime_error (
+        "queuerawtransaction \"hexstring\" (\"options\")\n"
+        "\nQueue a transaction for future broadcast.\n"
+        "\nArguments:\n"
+        "1. \"hexstring\"             (string, required) The hex string of the raw transaction\n"
+        "2. \"options\"               (object, optional)\n"
+        "   {\n"
+        "     \"sendWhen\"            (object, optional) Conditions upon which the transaction will be broadcast\n"
+        "     {\n"
+        "       \"txid\"              (string, optional) Transaction ID to watch for confirmation\n"
+        "       \"name\"              (string, optional) Name to watch for confirmation\n"
+        "       \"confirmations\"     (numeric, optional) Queued transaction will be broadcast when the txid or name (most recent update) has this many confirmations\n"
+        "     }\n"
+        "   }\n"
+        "\nResult:\n"
+        "\"txid\"             (string) the queued transaction's hash in hex\n"
+        // TODO: Examples
```

Comment:
Working on examples now.  Planning to supply examples for name registration, name renewal, and name sniping.

Question: if Alice has a confirmed `name_new` for a name that already exists and wants to register the name as soon as it expires, at what number of confirmations (for the most recent `name_update`) should she broadcast the `name_firstupdate`?  My loose guess is that the name is available to be re-registered in the block at which the name has 36k confirmations, thus she should try to broadcast at 36k-1 confirmations so that miners see it in the mempool when mining block 36k.  But I've never done this experiment, so not sure if I'm correct (and I'm not sure if the mempool will admit name registrations for names that currently have 36k-1 confirmations).